.github/workflows/uc_loadsim.yml

```diff
@@ -30,9 +37,528 @@ defaults:
+
+    - name: Prepare certificates from secrets
+      working-directory: wlan-testing/helm/ucentral
+      run: |
+        echo "${{ secrets.DIGICERT_CERT }}" | base64 -d > cert.pem
+        echo "${{ secrets.DIGICERT_KEY }}" | base64 -d > key.pem
+        echo "${{ secrets.LOADSIM_DIGICERT_CERT }}" | base64 -d > device-cert.pem
+        echo "${{ secrets.LOADSIM_DIGICERT_KEY }}" | base64 -d > device-key.pem
+
+    - name: Deploy CloudSDK instance with load simulator
+      working-directory: wlan-testing/helm/ucentral
+      run: |
+        export NAMESPACE="${{ steps.ec2_facts.outputs.instance_name }}"
+        export DEPLOY_METHOD=git
+        export CHART_VERSION=main
+        export OWGW_VERSION=master
+        export OWGWUI_VERSION=main
+        export OWSEC_VERSION=main
+        export OWFMS_VERSION=main
+        export OWPROV_VERSION=main
+        export OWPROVUI_VERSION=main
+        export OWANALYTICS_VERSION=main
+        export OWSUB_VERSION=main
+        export VALUES_FILE_LOCATION=values.ucentral-qa.yaml,values.ucentral-qa.test-nodes.yaml,values.ucentral-qa.owls-enabled.yaml
+        export OWGW_AUTH_USERNAME=${{ secrets.UCENTRALGW_AUTH_USERNAME }}
+        export OWGW_AUTH_PASSWORD=${{ secrets.UCENTRALGW_AUTH_PASSWORD }}
+        export OWFMS_S3_SECRET=${{ secrets.UCENTRALFMS_S3_SECRET }}
+        export OWFMS_S3_KEY=${{ secrets.UCENTRALFMS_S3_KEY }}
+        export CERT_LOCATION=cert.pem
+        export KEY_LOCATION=key.pem
+        export DEVICE_CERT_LOCATION=device-cert.pem
+        export DEVICE_KEY_LOCATION=device-key.pem
+        export OWSEC_NEW_PASSWORD=${{ secrets.OWSEC_NEW_PASSWORD }}
+        export USE_SEPARATE_OWGW_LB=true
+        ./deploy.sh
+
+    # Run tests
+    - name: Checkout OWLS for CLI tool
+      uses: actions/checkout@v2
+      with:
+        repository: Telecominfraproject/wlan-cloud-owls
+        path: wlan-cloud-owls
+
+    - name: Run simulation
+      working-directory: wlan-cloud-owls/test_scripts/curl
+      run: |
+        export OWGW="${{ steps.ec2_facts.outputs.instance_name }}.cicd.lab.wlan.tip.build:15002"
+        export OWSEC="${{ steps.sec.outputs.url }}:16001"
+        export OWLS="${{ steps.ls.outputs.url }}:16007"
+        echo "OWGW - $OWGW"
+        echo "OWSEC - $OWSEC"
+        echo "OWLS - $OWLS"
+
+        echo "Trying to cache OWLS FQDN in /etc/hosts"
+        export HOST_ENTRY=""
+        until [ ! -z "$HOST_ENTRY" ]; do export HOST_ENTRY=$(getent hosts ${{ steps.ls.outputs.url }}); sleep 1; done;
+        cat /etc/hosts > /tmp/hosts
+        echo "$HOST_ENTRY" >> /tmp/hosts
+        sudo cp /tmp/hosts /etc/hosts
+        echo "DNS record for $OWLS resolved successfully!"
+        cat /etc/hosts
+
+        curl -s -X POST -H 'Content-Type: application/json' https://$OWSEC/api/v1/oauth2 -d '{"userId": "tip@ucentral.com", "password": "'${{ secrets.OWSEC_NEW_PASSWORD }}'"}' | jq '.access_token' -r > access_token
```

Comment:
I'd suggest that switch to the OWLS CLI in the future instead of re-implementing this to avoid the need to keep track of breaking API changes.

Reply:
Yes, I guess I could move that to OWLS CLI so that this logic would be accessible in other scenarios and by other community members